osgp/platform/osgp-adapter-ws-smartmetering/src/main/java/org/opensmartgridplatform/adapter/ws/smartmetering/endpoints/SmartMeteringInstallationEndpoint.java

```diff
@@ -447,4 +470,72 @@ public DecoupleMbusDeviceByChannelResponse getDecoupleMbusDeviceByChannelRespons
         }
         return response;
     }
-}
+
+    @PayloadRoot(localPart = "SetSubscriptionInformationRequest", namespace = SMARTMETER_INSTALLATION_NAMESPACE)
+    @ResponsePayload
+    public SetSubscriptionInformationAsyncResponse setSubscriptionInformation(
+            @OrganisationIdentification final String organisationIdentification,
+            @RequestPayload final SetSubscriptionInformationRequest request,
+            @MessagePriority final String messagePriority, @ScheduleTime final String scheduleTime,
+            @ResponseUrl final String responseUrl) throws OsgpException {
+
+        log.info("Incoming SetSubscriptionInformation for meter: {}.", request.getDeviceIdentification());
+
+        SetSubscriptionInformationAsyncResponse response = null;
+        try {
+            response = new SetSubscriptionInformationAsyncResponse();
+
+            final SetSubscriptionInformationRequestData requestData = this.installationMapper.map(request,
+                    SetSubscriptionInformationRequestData.class);
+
+            final String correlationUid = this.installationService.enqueueSetSubscriptionInformationRequest(
+                    organisationIdentification, request.getDeviceIdentification(), requestData,
+                    MessagePriorityEnum.getMessagePriority(messagePriority),
+                    this.installationMapper.map(scheduleTime, Long.class));
+
+            response.setCorrelationUid(correlationUid);
+            response.setDeviceIdentification(request.getDeviceIdentification());
+            this.saveResponseUrlIfNeeded(correlationUid, responseUrl);
+
+        } catch (final ConstraintViolationException e) {
+
+            log.error("Exception: {} while setting subcription information for device: {} for organisation {}.",
+                    e.getMessage(), request.getDeviceIdentification(), organisationIdentification, e);
+
+            throw new FunctionalException(FunctionalExceptionType.VALIDATION_ERROR, ComponentType.WS_CORE,
+                    new ValidationException(e.getConstraintViolations()));
+
+        } catch (final Exception e) {
+
+            log.error("Exception: {} while updating device: {} for organisation {}.", e.getMessage(),
+                    request.getDeviceIdentification(), organisationIdentification, e);
+
+            this.handleException(e);
+        }
+        return response;
+    }
+
+    @PayloadRoot(localPart = "SetSubscriptionInformationAsyncRequest", namespace = SMARTMETER_INSTALLATION_NAMESPACE)
+    @ResponsePayload
+    public SetSubscriptionInformationResponse getSetSubscriptionInformationResponse(
```

Comment:
I'd like to check whether "SubscriptionInformation" is a desirable name for the information that is maintained in the newly added operations.
My feeling is that it has relevant meaning in some other context than GXF, which might also explain all the unused fields (MeId, Esn, UimId, EqId. Mdn, Status, CustCode and SupplierReferenceId), where only IpAddress, BtsId and CellId are used.
Would something like "NetworkingInformation" be a better name here, or do I need to acquaint myself a bit more with the meaning of "SubscriptionInformation"?

Reply:
this will be changed to SetCommunicationNetworkInformation

Reply:
While the class names have been renamed, there still is something remaining regarding the Subscription Information (the `exceptionContext` input `String` for the exception thrown if the result from setting the Communication Network Information is not OK).

Reply:
> While the class and method names etc. have been renamed, there still is something remaining regarding the Subscription Information (the `exceptionContext` input `String` for the exception thrown if the result from setting the Communication Network Information is not OK).

fixed.